src/main/java/org/folio/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -1,16 +1,111 @@
 package org.folio.orders.lines.update.instance;
 
-import org.apache.commons.lang.NotImplementedException;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.HttpException;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.folio.orders.lines.update.OrderLineUpdateInstanceHolder;
 import org.folio.orders.lines.update.OrderLineUpdateInstanceStrategy;
 import org.folio.rest.core.models.RequestContext;
 
 import io.vertx.core.Future;
+import org.folio.rest.jaxrs.model.Holding;
+import org.folio.rest.jaxrs.model.Location;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.jaxrs.model.ReplaceInstanceRef;
+import org.folio.rest.persist.DBClient;
+import org.folio.rest.persist.Tx;
+import org.folio.services.lines.PoLinesService;
+import org.folio.services.piece.PieceService;
+import org.folio.services.title.TitleService;
 
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RequiredArgsConstructor
 public class WithHoldingOrderLineUpdateInstanceStrategy implements OrderLineUpdateInstanceStrategy {
 
+  private final TitleService titleService;
+  private final PoLinesService poLinesService;
+  private final PieceService pieceService;
+  private final Logger logger = LogManager.getLogger(WithHoldingOrderLineUpdateInstanceStrategy.class);
+
   @Override
   public Future<Void> updateInstance(OrderLineUpdateInstanceHolder holder, RequestContext rqContext) {
-    throw new NotImplementedException();
+    Promise<Void> promise = Promise.promise();
+    if (holder.getPatchOrderLineRequest().getReplaceInstanceRef() == null
+        || holder.getPatchOrderLineRequest().getReplaceInstanceRef().getHoldings() == null) {
```

Comment:
`holder.getPatchOrderLineRequest().getReplaceInstanceRef().getHoldings()` never be null, but could be empty.
This list initiated according to schema
![image](https://user-images.githubusercontent.com/45555481/173026154-d131b090-66bd-4431-90d4-1796a6e556f5.png)